crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -926,14 +926,12 @@ impl<'db> TypeInferenceBuilder<'db> {
         } = with_statement;
 
         for item in items {
-            match item.optional_vars.as_deref() {
-                Some(ast::Expr::Name(name)) => {
-                    self.infer_definition(name);
-                }
-                _ => {
-                    // TODO infer definitions in unpacking assignment
-                    self.infer_expression(&item.context_expr);
-                }
+            if let Some(ast::Expr::Name(name)) = item.optional_vars.as_deref() {
+                self.infer_definition(name);
+            } else {
+                // TODO infer definitions in unpacking assignment
+                self.infer_expression(&item.context_expr);
+                self.infer_optional_expression(item.optional_vars.as_deref());
             }
```

Comment:
```suggestion
            let target = item.optional_vars.as_deref();
            if let Some(ast::Expr::Name(name)) = target {
                self.infer_definition(name);
            } else {
                // TODO infer definitions in unpacking assignment
                self.infer_expression(&item.context_expr);
                self.infer_optional_expression(target);
            }
```

Reply:
I thought about this and decided not to bother 😆 I would be very surprised if it makes any difference to the generated code. But sure, doesn't hurt, it's a bit nicer to read.